components/admin/AdminFilter.tsx

```diff
@@ -0,0 +1,155 @@
+import { FirebaseTablesEnum, StatusEnum } from "@/lib/enums";
+import { cn } from "@/lib/utils";
+import { Button } from "../ui/button";
+import { CheckIcon } from "@radix-ui/react-icons";
+import { Command, CommandEmpty, CommandGroup, CommandInput, CommandItem } from "@/components/ui/command";
+import { useState } from "react";
+import { Filter, getFilters } from "@/lib/api";
+import { Dialog, DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogTitle } from "../ui/dialog";
+import { Input } from "@/components/ui/input";
+
+interface AdminFilterProps {
+  filterTypeSingular: string;
+  filterTypePlural: string;
+  filterTable: FirebaseTablesEnum;
+  memberId: string;
+  filters: { name: string; id: string; status: string }[];
+  setFilters: (filter: { name: string; id: string; status: string }[]) => void;
+  emoji: string;
+  suggestedFilter: string;
+  setSuggestedFilter: (suggestedFilter: string) => void;
+}
+
+export default function AdminFilter({
+  filterTypeSingular,
+  filterTypePlural,
+  filterTable,
+  memberId,
+  filters,
+  setFilters,
+  emoji,
+  suggestedFilter,
+  setSuggestedFilter,
+}: AdminFilterProps) {
+  const [open, setOpen] = useState(false);
+  const [allFilters, setAllFilters] = useState<Filter[]>([]);
+  const [selectedFilters, setSelectedFilters] = useState<string[]>(filters ? filters.map((f) => f.id) : []);
+  const [suggestOpen, setSuggestOpen] = useState(false);
+  const unnapprovedFilters = filters ? filters.filter((f) => f.status !== StatusEnum.APPROVED).map((f) => f.name) : [];
+
+  const filterClass =
+    "inline-block rounded-xl border px-3 py-0.5 my-0.5 text-sm tracking-wide text-secondary-foreground";
+
+  const handleOpen = async () => {
+    setOpen(!open);
+
+    if (unnapprovedFilters.length > 0) {
+      setSuggestedFilter(unnapprovedFilters[0]);
+      setSuggestOpen(true);
+    } else if (suggestedFilter === "" || suggestedFilter === null) {
+      setSuggestOpen(false);
+    }
+    allFilters.length === 0 && setAllFilters(await getFilters(filterTable));
+  };
+
+  const handleSelect = (filter: Filter) => {
```

Comment:
Ahh, wait I'm just catching this now: let's do a quick check if 3 have already been selected.

Reply:
good catch, will add a check for this